module/zfs/zfs_acl.c

```diff
@@ -2418,12 +2421,13 @@ zfs_fastaccesschk_execute(znode_t *zdp, cred_t *cr)
 		return (0);
 	}
 
-	if (FUID_INDEX(zdp->z_uid) != 0 || FUID_INDEX(zdp->z_gid) != 0) {
+	if (zfs_uid_read(ZTOI(zdp)) != 0 ||
```

Comment:
So FUID_INDEX is `#define  FUID_INDEX(x)   ((x) >> 32)`, essentially taking the upper 32bit of the passed uid. This was correct when the 64bit z_uid were used, but now that the native 32 bit VFS members are used this wouldn't be correct as it would cause the UID to become 0.